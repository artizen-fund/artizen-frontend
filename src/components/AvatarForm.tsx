import styled from 'styled-components'
import { Button, Glyph } from '@components'
import { InvisiFileInput, rgba } from '@lib'
import { breakpoint, palette } from '@theme'

interface IAvatarForm {
  setFile: (input: File) => void
  initialState?: string
}

const AvatarForm = ({ setFile, initialState }: IAvatarForm) => {
  const IMAGE_UPLOAD_BUTTON_ID = 'avatarUploadImage'
  return (
    <Wrapper>
      <InvisiFileInput id={IMAGE_UPLOAD_BUTTON_ID} {...{ setFile, initialState }}>
        <DropZone>
          <Glyph glyph="face" />
        </DropZone>
      </InvisiFileInput>
      <Buttons>
        {/*
        <Button level={2} onClick={() => alert('derp')}>
          Take photo
        </Button>
        */}
        <Button level={2} htmlFor={IMAGE_UPLOAD_BUTTON_ID} outline>
          Upload photo
        </Button>
      </Buttons>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  position: relative;
  grid-area: avatarForm;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: stretch;
  gap: 20px;
`

const Buttons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
`

const DropZone = styled.div<{ preview?: string }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80px;
  height: 80px;

  @media only screen and (min-width: ${breakpoint.laptop}px) {
    width: 100px;
    height: 100px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 120px;
    height: 120px;
  }
  ${props =>
    props.preview &&
    `
    background-image: url(${props.preview});
    background-size: cover;
    background-position: center center;
    > * {
      display: none;
    }
  `}
  border: 2px dashed black;
  @media (prefers-color-scheme: dark) {
    border: 2px dashed ${rgba(palette.moon)};
  }
  border-radius: 9999px;
  transition: border-color 0.2s ease-in-out;
`

export default AvatarForm
