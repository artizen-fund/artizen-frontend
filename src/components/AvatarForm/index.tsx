import { useRef, useState } from 'react'
import styled from 'styled-components'
import { Button, Glyph } from '@components'
import { rgba, assert } from '@lib'
import { palette, breakpoint } from '@theme'

interface IAvatarForm {
  setFile: (input: File) => void
}

const AvatarForm = ({ setFile }: IAvatarForm) => {
  const fileref = useRef<HTMLInputElement>()
  const [preview, setPreview] = useState<string>()

  const pickFile = (e: React.ChangeEvent<HTMLInputElement>) => {
    const pickedFile = e.target.files?.[0]
    if (!pickedFile) return
    setFile(pickedFile)
    const reader = new FileReader()
    reader.onloadend = () => setPreview(reader?.result as string)
    reader.readAsDataURL(pickedFile)
  }

  return (
    <Wrapper>
      <DropZone {...{ preview }}>{!preview && <Glyph glyph="face" />}</DropZone>
      <Buttons>
        {/*
        <Button level={2} onClick={() => alert('derp')}>
          Take photo
        </Button>
        */}
        <Button level={2} htmlFor="avatarUploadInput" outline>
          Upload photo
          <FileInput type="file" onChange={pickFile} {...{ fileref }} id="avatarUploadInput" />
        </Button>
      </Buttons>
    </Wrapper>
  )
}

const FileInput = styled.input`
  appearance: none;
  width: 1px;
  height: 1px;
  opacity: 0;
`

const Wrapper = styled.div`
  grid-area: avatarForm;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: stretch;
  gap: 20px;
`

const Buttons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
`

const DropZone = styled.div<{ preview?: string }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80px;
  height: 80px;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    width: 100px;
    height: 100px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    width: 120px;
    height: 120px;
  }
  ${props =>
    props.preview &&
    `
    background-image: url(${props.preview});
    background-size: cover;
    background-position: center center;
  `}
  border: 2px dashed black;
  border-radius: 9999px;
  transition: border-color 0.2s ease-in-out;
`

export default AvatarForm
