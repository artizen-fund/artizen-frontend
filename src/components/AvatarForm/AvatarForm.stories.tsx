import AvatarForm from './'

export default {
  title: 'components/AvatarForm',
  component: AvatarForm,
  argTypes: {},
}

export const AvatarFormComponent = (props: any) => <AvatarForm {...props} />
