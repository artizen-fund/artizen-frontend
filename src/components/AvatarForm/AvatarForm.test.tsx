import renderer from 'react-test-renderer'
import AvatarForm from './'

describe('AvatarForm', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AvatarForm />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
