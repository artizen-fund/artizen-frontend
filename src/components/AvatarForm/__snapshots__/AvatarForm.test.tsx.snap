// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarForm renders correctly 1`] = `
<div
  className="sc-ad49e03d-0 bPHiXN"
>
  <input
    className="sc-15a98cbf-0 jkrKKd"
    id="avatarUploadImage"
    onChange={[Function]}
    type="file"
  />
  <div
    className="sc-ad49e03d-2 imeOES"
    onClick={[Function]}
  >
    <div
      className="sc-3ba9b100-0 hUkqvF"
    />
  </div>
  <div
    className="sc-ad49e03d-1 bzbruR"
  >
    <label
      className="sc-313ded62-3 gWwXif undefined "
      htmlFor="avatarUploadImage"
    >
      <span>
        Upload photo
      </span>
    </label>
  </div>
</div>
`;
